src/cephadm/cephadmlib/node_proxy/baseredfishsystem.py

```diff
@@ -0,0 +1,152 @@
+import concurrent.futures
+from .basesystem import BaseSystem
+from .redfish_client import RedFishClient
+from threading import Thread, Lock
+from time import sleep
+from .util import Logger, retry
+from typing import Dict, Any, List
+
+
+class BaseRedfishSystem(BaseSystem):
+    def __init__(self, **kw: Any) -> None:
+        super().__init__(**kw)
+        self.common_endpoints: List[str] = kw.get('common_endpoints', ['/Systems/System.Embedded.1',
+                                                                       '/UpdateService'])
+        self.chassis_endpoint: str = kw.get('chassis_endpoint', '/Chassis/System.Embedded.1')
+        self.log = Logger(__name__)
+        self.host: str = kw['host']
+        self.port: int = kw['port']
+        self.username: str = kw['username']
+        self.password: str = kw['password']
+        # move the following line (class attribute?)
+        self.client: RedFishClient = RedFishClient(host=self.host, port=self.port, username=self.username, password=self.password)
+        self.log.logger.info(f'redfish system initialization, host: {self.host}, user: {self.username}')
+
+        self.run: bool = False
+        self.thread: Thread
+        self.data_ready: bool = False
+        self.previous_data: Dict = {}
+        self.lock: Lock = Lock()
+        self.data: Dict[str, Dict[str, Any]] = {}
+        self._system: Dict[str, Dict[str, Any]] = {}
+        self._sys: Dict[str, Any] = {}
+        self.start_client()
+
+    def start_client(self) -> None:
+        self.client.login()
+        self.start_update_loop()
+
+    def start_update_loop(self) -> None:
+        self.run = True
+        self.thread = Thread(target=self.update)
+        self.thread.start()
+
+    def stop_update_loop(self) -> None:
+        self.run = False
+        self.thread.join()
+
+    def update(self) -> None:
+        #  this loop can have:
+        #  - caching logic
+        while self.run:
+            self.log.logger.debug('waiting for a lock.')
+            self.lock.acquire()
+            self.log.logger.debug('lock acquired.')
```

Comment:
I know they're debug logs, but given there is a global cephadm lock as well, it would be nice if these were more specific about what lock they're waiting on

Reply:
node-proxy's logger sets the format so these messages would be like this:

```
2023-12-18 09:50:10,766 - cephadmlib.node_proxy.baseredfishsystem - DEBUG - waiting for a lock.
2023-12-18 09:50:19,912 - cephadmlib.node_proxy.baseredfishsystem - DEBUG - lock acquired.
```
I would think this is enough clear that node-proxy generates those messages.

Reply:
updated the messages